app/src/main/java/com/spascoding/taskycourse/MainViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package com.spascoding.taskycourse
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.spascoding.taskycourse.feature_register_screen.data.local.model.UserInfo
+import com.spascoding.taskycourse.feature_register_screen.data.local.model.UserInfoManager
+import com.spascoding.taskycourse.ui.theme.MainState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class MainViewModel @Inject constructor(
+    private var userInfoManager: UserInfoManager
+): ViewModel() {
+
+    var state = MutableStateFlow(MainState(
+        isReady = false,
+        isAuthenticated = false,
+    ))
+        private set
```

Comment:
The recommendation I shared only works for Compose state, not for StateFlow. The `var` will prevent the whole StateFlow being replaced from another class (`viewModel.state = MutableStateFlow(...)`), but not the StateFlow's value being replaced (`viewModel.state.value = ...`). Here you need two separate properties or you expose Compose state directly.